### ISSUE TITLE
SQLite: Can't parse DEFAULT keyword in ALTER TABLE statment

### ISSUE THREAD

**Issue by mtlynch** [sqlfluff/sqlfluff#3980]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

sqlfluff seems unable to parse the `DEFAULT` keyword in a SQLite `ALTER TABLE` query.

The `DEFAULT` seems to be a valid keyword according to the SQLite docs and this test script:

```bash
$ cat test.sql 
CREATE TABLE IF NOT EXISTS users(
    id INTEGER PRIMARY KEY AUTOINCREMENT,
    username TEXT NOT NULL UNIQUE
);

INSERT INTO users (username) VALUES ('alice'), ('bob');

ALTER TABLE users
ADD COLUMN credentials_last_changed
TEXT NOT NULL DEFAULT "0001-01-01T00:00:00.000000+00:00";

SELECT username, credentials_last_changed FROM users;

$ sqlite3 test.db < test.sql 
alice|0001-01-01T00:00:00.000000+00:00
bob|0001-01-01T00:00:00.000000+00:00
```

### Expected Behaviour

```bash
echo 'ALTER TABLE users
ADD COLUMN credentials_last_changed
TEXT NOT NULL DEFAULT "0001-01-01T00:00:00.000000+00:00";' | sqlfluff lint --dialect sqlite -
All Finished 📜 🎉!
```

### Observed Behaviour

```bash
$ echo 'ALTER TABLE users
ADD COLUMN credentials_last_changed
TEXT NOT NULL DEFAULT "0001-01-01T00:00:00.000000+00:00";' | sqlfluff lint --dialect sqlite -
== [stdin] FAIL                                                                                                                                                                             
L:   3 | P:  15 |  PRS | Line 3, Position 15: Found unparsable section: 'DEFAULT
                       | "0001-01-01T00:00:00.000000+00:0...'
WARNING: Parsing errors found and dialect is set to 'sqlite'. Have you configured your dialect correctly?
All Finished 📜 🎉!
```

### How to reproduce

```bash
echo 'ALTER TABLE users
ADD COLUMN credentials_last_changed
TEXT NOT NULL DEFAULT "0001-01-01T00:00:00.000000+00:00";' | sqlfluff lint --dialect sqlite -
```

### Dialect

SQLite

### Version

```bash
$ sqlfluff --version
sqlfluff, version 1.3.2
$ python --version
Python 3.9.14
```

### Configuration

None, just using the Docker container

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by Enduriel** [comment]
This is related to #5732, which is more specific and precise, there are problems in this issue
1) SQLite statements with ALTER TABLE are not parsed as in #5732 
2) Incorrect double quoting of `DEFAULT "0001-01-01T00:00:00.000000+00:00"`

Therefore this should be close in favor of #5732

**Comment by keraion** [comment]
Fixed by #5888
